### PR TITLE
CRUDs for Models

### PR DESCRIPTION
**Pull request by EndritMakolli**
Create, Read, Update and Delete Methods and Controllers added for each Model

### CODE REVIEW

File: API/Controllers/FacultiesController.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace API.Controllers
+{
+    [AllowAnonymous]
+    [Route("api/[controller]")]
+    [ApiController]
+
+    public class FacultiesController :  ControllerBase
+    {
+        private readonly IMediator _mediator;
```
Comment (rilindbicaj) [inline API/Controllers/FacultiesController.cs:18]: This is not necessary. You're already inheriting the Mediator attribute from BaseController.

File: Application/Faculties/Create.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            private readonly FacultyDBContext _context;
+
+            public Handler(FacultyDBContext context)
+            {
+                _context = context;
+            }
+
+            public async Task<Unit> Handle(Command request, CancellationToken cancellationToken)
+            {
+                var faculty = new Faculty
```
Comment (rilindbicaj) [inline Application/Faculties/Create.cs:39]: You've already provided a Faculty object, there's no need to create it again.

And again, navigation properties should not be included in request objects.

File: Application/Faculties/Edit.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                _context = context;
+            }
+        
+            public async Task<Unit> Handle(Command request, CancellationToken cancellationToken)
+            {
+                var faculty = await _context.Faculties.FindAsync(request.FacultyID);
+
+                if(faculty == null)
+                    throw new Exception ("Could not find faculty");
+
```
Comment (rilindbicaj) [inline Application/Faculties/Edit.cs:41]: Use automapper for this.

**Review by rilindbicaj — CHANGES_REQUESTED** [review]
- [x] Rewrite controllers - if a controller extends BaseController, they shouldn't need their own mediator - they already inherit one from BaseController. Remove all _mediator attributes and use the inherited Mediator from BaseController. Also, remove the constructors from the controllers. They're unnecessary.
- [x]  Use DTOs - And only add the necessary attributes to create the object. Navigation properties, such as "Major" and "FacultySemesters" are part of the domain model only. Also return response DTOs, not the entire model. Check SPMSPosts for examples.
- [x] Use Automapper - After implementing DTOs, use Automapper to map between them. 
- [x] Refactor redundant code - Automapper will handle the Edit methods, where most of the repeated code is.
- [x] Rename Application classes - Make the classes represent an actual use case instead of just Edit, Create, Delete.

**Comment by EndritMakolli** [comment]
Bic ti qr kap heki kta requesta, asnjana nuk o tu u bo inherit prej
basecontroller, ControllerBase so i njejti sen me BaseController, kshtuqe
osht e nevojshme me inherite qata, bone merge me main ma shpejt qet branch,
senet qe funsionojn munen mu ristrukturu edhe prej main edhe qe tdoket
najsen “unnecessary” ma shpejt ja bon delete se me ma shkru kete koment.

Me respekt Endriti

**Comment by rilindbicaj** [comment]
image

**Pull request merged**